Core/NoOpRunner.Core/Shapes/Player.cs

```diff
@@ -215,6 +215,11 @@ public PlayerOneState State
             private set => StateMachine.State = value;
         }
 
+        public void DoAction()
+        {
+            StateMachine.DoAction();
+        }
```

Comment:
Don't litter the codebase like this... Just add a log where the state pattern is actually being used, maybe in `StateMachine` where you change the state

Reply:
Fixed